### ISSUE TITLE
New build error with interactive script

### ISSUE THREAD

**Issue by ppebay** [DARMA-tasking/vt-tv#123]
## Steps to reproduce:
Run ./interactive_build.sh; or use the `pip install .` approach, which yields similar errors.

## Description of the issue:
```
vt-tv$ ./interactive_build.sh 
Welcome to the vt-tv Build Setup!
Enter the vt-tv source directory [default (Enter): /Users/pppebay/Documents/Git/vt-tv]: 
Enter the build directory [default (Enter): /Users/pppebay/Documents/Git/vt-tv/build]: 
Do you have a precompiled VTK directory? (y/n): y
Enter the VTK build directory : /Users/pppebay/Documents/Build/VTK-9.3.1@Release/
Enter the number of jobs/threads for make (leave empty for system default): 5
Do you want to install the project after building? (y/n): n
```
...
```
In file included from /Users/pppebay/Documents/Git/vt-tv/tests/unit/api/test_object_info.cc:46:
/Users/pppebay/Documents/Git/vt-tv/tests/unit/api/../util.h:87:16: error: cannot take the address of an rvalue of type 'int'
      status = WEXITSTATUS(pclose(pipe));
               ^           ~~~~~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:144:27: note: expanded from macro 'WEXITSTATUS'
#define WEXITSTATUS(x)  ((_W_INT(x) >> 8) & 0x000000ff)
                          ^      ~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:131:34: note: expanded from macro '_W_INT'
#define _W_INT(w)       (*(int *)&(w))  /* convert union wait to int */
                                 ^ ~
In file included from /Users/pppebay/Documents/Git/vt-tv/tests/unit/api/test_object_info.cc:46:
/Users/pppebay/Documents/Git/vt-tv/tests/unit/api/../util.h:90:14: error: cannot take the address of an rvalue of type 'int'
    status = WEXITSTATUS(pclose(pipe));
             ^           ~~~~~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:144:27: note: expanded from macro 'WEXITSTATUS'
#define WEXITSTATUS(x)  ((_W_INT(x) >> 8) & 0x000000ff)
                          ^      ~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:131:34: note: expanded from macro '_W_INT'
#define _W_INT(w)       (*(int *)&(w))  /* convert union wait to int */
                                 ^ ~
/Users/pppebay/Documents/Git/vt-tv/tests/unit/api/test_object_info.cc:55:25: error: no matching constructor for initialization of 'ObjectInfo'
  ObjectInfo object_0 = ObjectInfo(
                        ^
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:70:3: note: candidate constructor not viable: no known conversion from 'vector<size_t>' to 'const vector<UniqueIndexBitType>' for 4th argument
  ObjectInfo(
  ^
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:59:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 4 were provided
struct ObjectInfo {
       ^
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:59:8: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 4 were provided
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:60:3: note: candidate constructor not viable: requires 0 arguments, but 4 were provided
  ObjectInfo() = default;
  ^
/Users/pppebay/Documents/Git/vt-tv/tests/unit/api/test_object_info.cc:61:25: error: no matching constructor for initialization of 'ObjectInfo'
  ObjectInfo object_1 = ObjectInfo(
                        ^
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:70:3: note: candidate constructor not viable: no known conversion from 'vector<size_t>' to 'const vector<UniqueIndexBitType>' for 4th argument
  ObjectInfo(
  ^
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:59:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 4 were provided
struct ObjectInfo {
       ^
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:59:8: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 4 were provided
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:60:3: note: candidate constructor not viable: requires 0 arguments, but 4 were provided
  ObjectInfo() = default;
  ^
In file included from /Users/pppebay/Documents/Git/vt-tv/tests/unit/api/test_object_communicator.cc:46:
/Users/pppebay/Documents/Git/vt-tv/tests/unit/api/../util.h:87:16: error: cannot take the address of an rvalue of type 'int'
      status = WEXITSTATUS(pclose(pipe));
               ^           ~~~~~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:144:27: note: expanded from macro 'WEXITSTATUS'
#define WEXITSTATUS(x)  ((_W_INT(x) >> 8) & 0x000000ff)
                          ^      ~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:131:34: note: expanded from macro '_W_INT'
#define _W_INT(w)       (*(int *)&(w))  /* convert union wait to int */
                                 ^ ~
In file included from /Users/pppebay/Documents/Git/vt-tv/tests/unit/api/test_object_communicator.cc:46:
/Users/pppebay/Documents/Git/vt-tv/tests/unit/api/../util.h:90:14: error: cannot take the address of an rvalue of type 'int'
    status = WEXITSTATUS(pclose(pipe));
             ^           ~~~~~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:144:27: note: expanded from macro 'WEXITSTATUS'
#define WEXITSTATUS(x)  ((_W_INT(x) >> 8) & 0x000000ff)
                          ^      ~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:131:34: note: expanded from macro '_W_INT'
#define _W_INT(w)       (*(int *)&(w))  /* convert union wait to int */
                                 ^ ~
In file included from /Users/pppebay/Documents/Git/vt-tv/tests/unit/api/test_info.cc:47:
/Users/pppebay/Documents/Git/vt-tv/tests/unit/api/../util.h:87:16: error: cannot take the address of an rvalue of type 'int'
      status = WEXITSTATUS(pclose(pipe));
               ^           ~~~~~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:144:27: note: expanded from macro 'WEXITSTATUS'
#define WEXITSTATUS(x)  ((_W_INT(x) >> 8) & 0x000000ff)
                          ^      ~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:131:34: note: expanded from macro '_W_INT'
#define _W_INT(w)       (*(int *)&(w))  /* convert union wait to int */
                                 ^ ~
In file included from /Users/pppebay/Documents/Git/vt-tv/tests/unit/api/test_info.cc:47:
/Users/pppebay/Documents/Git/vt-tv/tests/unit/api/../util.h:90:14: error: cannot take the address of an rvalue of type 'int'
    status = WEXITSTATUS(pclose(pipe));
             ^           ~~~~~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:144:27: note: expanded from macro 'WEXITSTATUS'
#define WEXITSTATUS(x)  ((_W_INT(x) >> 8) & 0x000000ff)
                          ^      ~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:131:34: note: expanded from macro '_W_INT'
#define _W_INT(w)       (*(int *)&(w))  /* convert union wait to int */
                                 ^ ~
4 errors generated.
2 errors generated.
make[2]: *** [tests/unit/CMakeFiles/AllTests.dir/api/test_object_info.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [tests/unit/CMakeFiles/AllTests.dir/api/test_object_communicator.cc.o] Error 1
In file included from /Users/pppebay/Documents/Git/vt-tv/tests/unit/api/test_info.cc:48:
/Users/pppebay/Documents/Git/vt-tv/tests/unit/api/../generator.h:133:32: error: no matching constructor for initialization of 'ObjectInfo'
      ObjectInfo object_info = ObjectInfo(it.first, 0, migratable, idx);
                               ^          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:70:3: note: candidate constructor not viable: no known conversion from 'vector<size_t>' to 'const vector<UniqueIndexBitType>' for 4th argument
  ObjectInfo(
  ^
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:59:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 4 were provided
struct ObjectInfo {
       ^
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:59:8: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 4 were provided
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:60:3: note: candidate constructor not viable: requires 0 arguments, but 4 were provided
  ObjectInfo() = default;
  ^
/Users/pppebay/Documents/Git/vt-tv/tests/unit/api/test_info.cc:185:23: error: no matching constructor for initialization of 'ObjectInfo'
  ObjectInfo o_info = ObjectInfo(0, 0, true, idx);
                      ^          ~~~~~~~~~~~~~~~
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:70:3: note: candidate constructor not viable: no known conversion from 'vector<size_t>' to 'const vector<UniqueIndexBitType>' for 4th argument
  ObjectInfo(
  ^
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:59:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 4 were provided
struct ObjectInfo {
       ^
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:59:8: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 4 were provided
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:60:3: note: candidate constructor not viable: requires 0 arguments, but 4 were provided
  ObjectInfo() = default;
  ^
In file included from /Users/pppebay/Documents/Git/vt-tv/tests/unit/api/test_object_work.cc:46:
/Users/pppebay/Documents/Git/vt-tv/tests/unit/api/../util.h:87:16: error: cannot take the address of an rvalue of type 'int'
      status = WEXITSTATUS(pclose(pipe));
               ^           ~~~~~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:144:27: note: expanded from macro 'WEXITSTATUS'
#define WEXITSTATUS(x)  ((_W_INT(x) >> 8) & 0x000000ff)
                          ^      ~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:131:34: note: expanded from macro '_W_INT'
#define _W_INT(w)       (*(int *)&(w))  /* convert union wait to int */
                                 ^ ~
In file included from /Users/pppebay/Documents/Git/vt-tv/tests/unit/api/test_object_work.cc:46:
/Users/pppebay/Documents/Git/vt-tv/tests/unit/api/../util.h:90:14: error: cannot take the address of an rvalue of type 'int'
    status = WEXITSTATUS(pclose(pipe));
             ^           ~~~~~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:144:27: note: expanded from macro 'WEXITSTATUS'
#define WEXITSTATUS(x)  ((_W_INT(x) >> 8) & 0x000000ff)
                          ^      ~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/wait.h:131:34: note: expanded from macro '_W_INT'
#define _W_INT(w)       (*(int *)&(w))  /* convert union wait to int */
                                 ^ ~
In file included from /Users/pppebay/Documents/Git/vt-tv/tests/unit/api/test_object_work.cc:47:
/Users/pppebay/Documents/Git/vt-tv/tests/unit/api/../generator.h:133:32: error: no matching constructor for initialization of 'ObjectInfo'
      ObjectInfo object_info = ObjectInfo(it.first, 0, migratable, idx);
                               ^          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:70:3: note: candidate constructor not viable: no known conversion from 'vector<size_t>' to 'const vector<UniqueIndexBitType>' for 4th argument
  ObjectInfo(
  ^
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:59:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 4 were provided
struct ObjectInfo {
       ^
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:59:8: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 4 were provided
/Users/pppebay/Documents/Git/vt-tv/src/vt-tv/api/object_info.h:60:3: note: candidate constructor not viable: requires 0 arguments, but 4 were provided
  ObjectInfo() = default;
  ^
4 errors generated.
make[2]: *** [tests/unit/CMakeFiles/AllTests.dir/api/test_info.cc.o] Error 1
3 errors generated.
make[2]: *** [tests/unit/CMakeFiles/AllTests.dir/api/test_object_work.cc.o] Error 1
make[1]: *** [tests/unit/CMakeFiles/AllTests.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 94%] Linking CXX executable vt-tv_standalone
[ 94%] Built target vt-tv_standalone
```

Despite the errors above, the final message is nonetheless the following:
```
[ 94%] Built target vt-tv_standalone
make: *** [all] Error 2
vt-tv built successfully!
```
which seems to contradict the above.

When using the `pip install .` approach, similar errors are found, but with the following final message:
```
      3 errors generated.
      make[2]: *** [tests/unit/CMakeFiles/AllTests.dir/api/test_object_work.cc.o] Error 1
      3 errors generated.
      make[2]: *** [tests/unit/CMakeFiles/AllTests.dir/api/test_phase_work.cc.o] Error 1
      4 errors generated.
      make[2]: *** [tests/unit/CMakeFiles/AllTests.dir/api/test_info.cc.o] Error 1
      3 errors generated.
      make[2]: *** [tests/unit/CMakeFiles/AllTests.dir/api/test_rank.cc.o] Error 1
      make[1]: *** [tests/unit/CMakeFiles/AllTests.dir/all] Error 2
      make[1]: *** Waiting for unfinished jobs....
      1 warning generated.
      [ 94%] Linking CXX shared module /Users/pppebay/Documents/Git/vt-tv/build/lib.macosx-11.1-arm64-cpython-39/vttv.cpython-39-darwin.so
      [ 94%] Built target vttv
      [ 95%] Linking CXX executable vt-tv_standalone
      [ 95%] Built target vt-tv_standalone
      make: *** [all] Error 2
      error: command '/opt/homebrew/bin/cmake' failed with exit code 2
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for vttv
  Running setup.py clean for vttv
Failed to build vttv
ERROR: Could not build wheels for vttv, which is required to install pyproject.toml-based projects
```

**Comment by tlamonthezie** [comment]
@ppebay This is an error with CLang. It is fixed in #97

**Comment by ppebay** [comment]
Problem disappears indeed with branch #97 

We should merge it ASAP @tlamonthezie

**Comment by ppebay** [comment]
Fixed by #110
